### ISSUE TITLE
large diffs are confusing

### ISSUE THREAD

**Issue by mixmix** [tbeseda/tap-spek#25]
not the worst in this case
![image](https://user-images.githubusercontent.com/2665886/159874001-f12fc240-109e-42f1-8924-a2f4950e6e8e.png)

gets worse when keys are in a different order (no deterministic ordering)
![image](https://user-images.githubusercontent.com/2665886/159874299-1d58caa8-d7ea-4edb-9211-06c23fe4b5b3.png)

:fire: totally explodes when the diff is "below the fold" (in this case inside one of those Array that is collapsed)
![image](https://user-images.githubusercontent.com/2665886/159874551-bba2d6ab-91ae-404a-adf0-53b002a7e970.png)

**Comment by mixmix** [comment]
```js
t.deepEqual(
  {
    id: 'Grandad',
    children: [
      {
        id: 'Dad',
        children: [
          {
            id: 'Me',
            children: ['FIRE'] // <<<<<<
          }
        ]
      },
      {
        id: 'Aunt',
        children: []
      }
    ]
  },
  {
    id: 'Grandad',
    children: [
      {
        id: 'Dad',
        children: [
          {
            id: 'Me',
            children: []
          }
        ]
      },
      {
        id: 'Aunt',
        children: []
      }
    ]
  }
)
```

**Comment by mixmix** [comment]
I totally appreciate that diffing is a hard problem. Really appreciate the work y'all have put into making this nice module, and how responsive you were to my last PR

:heart: :floppy_disk: :turtle:

**Comment by mixmix** [comment]
woah I followed that a fair way down the rabbit hole.

The problem is here:
```js
parser.on('fail', (fail) => {
  print(`${pad(2)}${FAIL} ${dim(`${fail.id})`)} ${red(fail.name)}\n`)

  if (fail.diag) {
    const { actual, at, expected, operator, stack } = fail.diag
    let msg = [] // individual lines of output

    if ([ 'equal', 'deepEqual' ].includes(operator)) {
      if (typeof expected === 'string' && typeof actual === 'string') {
        msg = [ ...msg, ...makeDiff(actual, expected) ]  // <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
      }

```
We end up on this line (see <<<), which means the `fail.diag.actual` for is a string... a stringified version of the object, but with `[Array]` or `[Object]` when the object gets too deep.
where does `fail` come from?

couldn't quite follow it.. went through `tap-parser`, couldn't follow it, but figured `tape` is the stream source... but didn't have much luck reading that either. kinda intense code!

**Comment by tbeseda** [comment]
Thanks for the sample assertion, I can create a new test for this. I'm clearly not testing large enough objects 😬 

~~I think the solve for large objects is to have `tcompare` use the `includeEnumerable` option. I will test that out today.~~

Ah, I see what you were saying about "below the fold". The TAP report from `tape` doesn't print out the full object so when `tap-arc` gets to it, the objects appear to be the same.
`tap-arc` can't really tell you what is wrong, but it shouldn't totally blow up since we do know it is a failure. It should print a generic error.

[btw you can use 0.2.0 to get the full stringified diffs]

**Comment by mixmix** [comment]
You're totally right, it's `tape` . I found a way to fix it:

```bash
$ NODE_TAPE_OBJECT_PRINT_DEPTH=10 tape 'test/*.test.mjs'"
```

or 

```js
test('my test name', { objectPrintDepth: 10 }, t => { })
```

![image](https://user-images.githubusercontent.com/2665886/160019595-aa92d52a-6f99-4605-9f66-44195127f196.png)

weird that it drops the rest of the object... not sure if that's `tap-arc` doing that or tape...

**Comment by tbeseda** [comment]
@mixmix Oh that's a good find.
I did also find that I was using `tcompare`'s `match` method where `strict` would provide more complete diffs. So I think it was a combinations of shallow TAP output and `tap-arc` excluding some parts of the diff.

0.3.2 fixed the bug you pointed out in the third screenshot in the original issue and tap-arc is now set up to be more verbose with diffs.
I've also fixed diff display when there are multiple "chunks" in a single diff. That will get released in 0.3.3. 

I'm going to close this issue with that release and if there are further ideas about improving larger diffs, we can open a new issue.

**Comment by mixmix** [comment]
sounds perfect @tbeseda 
once again thanks for y'alls responsiveness, it's really nice collaborating on tools we can all share

that security warning on tap-spec was grating on me, and lack of diff was annoying. tap-diff was ok but not deterministic key ordering (I forked that to add it).

do you use tape or something else to write tests?
